test/TransitivePathTest.cpp

```diff
@@ -50,29 +50,95 @@ class TransitivePathTest : public testing::TestWithParam<bool> {
+    return T->bindLeftSide(leftOp, sideTableCol);
+  }
+
+  // ___________________________________________________________________________
   [[nodiscard]] static std::shared_ptr<TransitivePathBase> makePathRightBound(
       IdTable input, Vars vars, IdTable sideTable, size_t sideTableCol,
       Vars sideVars, TransitivePathSide left, TransitivePathSide right,
-      size_t minDist, size_t maxDist) {
+      size_t minDist, size_t maxDist, bool forceFullyMaterialized) {
+    auto [T, qec] = makePath(std::move(input), vars, std::move(left),
+                             std::move(right), minDist, maxDist);
+    auto rightOp = ad_utility::makeExecutionTree<ValuesForTesting>(
+        qec, std::move(sideTable), sideVars, false,
+        std::vector<ColumnIndex>{sideTableCol}, LocalVocab{}, std::nullopt,
+        forceFullyMaterialized);
+    return T->bindRightSide(rightOp, sideTableCol);
+  }
+
+  // ___________________________________________________________________________
+  [[nodiscard]] static std::shared_ptr<TransitivePathBase> makePathRightBound(
+      IdTable input, Vars vars, std::vector<IdTable> sideTables,
+      size_t sideTableCol, Vars sideVars, TransitivePathSide left,
+      TransitivePathSide right, size_t minDist, size_t maxDist) {
     auto [T, qec] = makePath(std::move(input), vars, std::move(left),
                              std::move(right), minDist, maxDist);
     auto rightOp = ad_utility::makeExecutionTree<ValuesForTesting>(
-        qec, std::move(sideTable), sideVars);
+        qec, std::move(sideTables), sideVars, false,
+        std::vector<ColumnIndex>{sideTableCol});
     return T->bindRightSide(rightOp, sideTableCol);
   }
+
+  // ___________________________________________________________________________
+  static std::vector<IdTable> split(const IdTable& idTable) {
+    std::vector<IdTable> result;
+    for (const auto& row : idTable) {
+      result.emplace_back(idTable.numColumns(), idTable.getAllocator());
+      result.back().push_back(row);
+    }
+    return result;
+  }
+
+  // ___________________________________________________________________________
+  static bool requestLaziness() { return std::get<1>(GetParam()); }
+
+  // ___________________________________________________________________________
+  void assertResultMatchesIdTable(const Result& result, const IdTable& expected,
+                                  ad_utility::source_location loc =
+                                      ad_utility::source_location::current()) {
+    auto t = generateLocationTrace(loc);
+    using ::testing::UnorderedElementsAreArray;
+    ASSERT_NE(result.isFullyMaterialized(), requestLaziness());
+    if (requestLaziness()) {
+      IdTable aggregateTable{expected.numColumns(), expected.getAllocator()};
+      std::vector<LocalVocab> localVocabs;
+      for (auto& [idTable, localVocab] : result.idTables()) {
+        localVocabs.emplace_back(std::move(localVocab));
+        aggregateTable.insertAtEnd(idTable);
```

Comment:
This pattern can also be used in other tests for laziness, so add it into one of our testing headers (maybe `IdTableHelpers` (of course with a short comment).